lib/takeaway.rb

```diff
@@ -0,0 +1,27 @@
+require_relative "menu"
+require_relative "sms"
+
+class Takeaway
+
+  attr_reader :menu, :order
+
+  def initialize(menu)
+    @menu = menu
+    @order = {}
+  end
+
+  def place_order(*order, total)
+    raise "The total is not correct and the order has not been placed!" if total != get_total(order)
+    @order = format_order(order)
+  end
+
+  def get_total(order)
+    format_order(order).map { |item, quantity| quantity * menu.dishes[item] }.inject(0, :+)
```

Comment:
I find this a little hard to read because there's quite a lot going on in one line.  What do you think of this alternative?

```ruby
def get_total(order)
  formatted_order = format(order)
  sub_totals = formatted_order.map { |dish, quantity| quantity * price_for(dish) }
  sub_totals.sum
end

def price_for(dish)
  menu.dishes[dish]
end
```

What do you think?  Is it easier to read?  Can you see any way of making it even more readable?